### PR TITLE
Feat delegation error success handling

### PR DESCRIPTION
**Pull request by cmancrypto**

### CODE REVIEW

**Comment by cmancrypto** [comment]
txHash doesn't seem to propagate through to success page. Would like to fix.

**Comment by cmancrypto** [comment]
Also would like to add a "isLoading" so that it looks like tx is doing something.

**Comment by dblackstone1** [comment]
> Also would like to add a "isLoading" so that it looks like tx is doing something.

there's a Loader that can be used.  example is in the button on the send page

**Comment by cmancrypto** [comment]
> > Also would like to add a "isLoading" so that it looks like tx is doing something.
> 
> there's a Loader that can be used. example is in the button on the send page

That should be in now

File: src/helpers/claimRewards.ts [hunk 1]
```diff
@@ -90,17 +90,48 @@ export const claimRewards = async (
...
 // Function to claim rewards and restake for one or multiple validators
-export const claimAndRestake = async (delegations: DelegationResponse | DelegationResponse[], rewards?: {
-  validator: string;
-  rewards: { denom: string; amount: string }[];
-}[]) => {
+export const claimAndRestake = async (
+  delegations: DelegationResponse | DelegationResponse[],
+  rewards?: {
+    validator: string;
+    rewards: { denom: string; amount: string }[];
```
Comment (dblackstone1) [inline src/helpers/claimRewards.ts:132]: update to the ValidatorReward interface could be good for this

**Comment by dblackstone1** [comment]
think i may need to run this to see what's fully going on with the xml changes

File: src/helpers/claimRewards.ts (hunk 1 above)
Comment (dblackstone1) [reply src/helpers/claimRewards.ts:132]: Thought you were referring to a bug with this though.  I'm not sure this action will be possible for all Cosmos chains.  Will be worth testing, but may need to be configured.  I know "claim all" can be done for any delegation for all of the chains though.

**Comment by dblackstone1** [comment]
This is off a single claim.  Problem with a filter function

Screenshot 2024-10-28 at 2 16 24 PM

**Comment by dblackstone1** [comment]
From what I saw it worked well though.  I was able to change off from the screen and view another, then come back to it and still see the loading.

A potential change is adding that as a field to the validator information, so each can have a separate state for the case of the long queries.  Or having each recorded separately so you can re-stake then immediately go to stake more on top of that and be able to take both actions.  Something to think about.  Could be a nice upgrade for later.

**Comment by cmancrypto** [comment]
so the filter issue should be resolved, at least I couldn't get it to recur. 

Ill add the validator loading state to issues so that I can get at that one later - that would be handy as a nice to have.

**Review by dblackstone1 — APPROVED** [review]

**Pull request merged**